osutil/user.go

```diff
@@ -150,12 +151,14 @@ func EnsureUserGroup(name string, id uint32, extraUsers bool) error {
 			delCmdStr = append(delCmdStr, "--extrausers")
 		}
 
-		// TODO: groupdel doesn't currently support --extrausers, so
-		// don't try to clean up when it is specified (LP: #1840375)
-		if !extraUsers {
-			delCmdStr = append(delCmdStr, name)
-			cmd = exec.Command(delCmdStr[0], delCmdStr[1:]...)
-			if output2, err2 := cmd.CombinedOutput(); err2 != nil {
+		delCmdStr = append(delCmdStr, name)
+		cmd = exec.Command(delCmdStr[0], delCmdStr[1:]...)
+		if output2, err2 := cmd.CombinedOutput(); err2 != nil {
+			if extraUsers && bytes.Contains(output2, []byte("unrecognized option")) {
+				// We have hit https://bugs.launchpad.net/bugs/1840375
+				// TODO: decide what to do.
```

Comment:
probably IMHO we should just log an error and return nil. The previous code gave no indication anything was wrong, whereas if we at least log the user will see that some part of the group deletion failed.